include/eve/arch/cpu/wide.hpp

```diff
@@ -121,59 +122,226 @@ namespace eve
     EVE_FORCEINLINE wide operator--(int)  noexcept  { auto that(*this); operator--(); return that; }
 
     //==============================================================================================
-    // Compound operators
+    // Bitwise operators
+    //==============================================================================================
+    friend EVE_FORCEINLINE auto operator~(wide const& v) noexcept
+    {
+      return detail::self_bitnot(v);
+    }
+
+    friend  EVE_FORCEINLINE auto operator&=(wide& w, value auto o) noexcept
+                        ->  decltype(detail::self_bitand(w, o))
```

Comment:
Why not just auto? We don't deduce type anywhere else?

Also I believe that `operator&=` and such should return a reference

Reply:
self_bitand has non-trivial constraints that I didn't want to be replicated here. So we use auto -> decltype to SFINAE out the operator if self_bitand can't be called. All the operators are merely trampolining to the proper function, hence the auto -> decltype.

As per &, self_bitand returns a ref, so the decltype is properly computed.